fapi-client/src/main/scala/com/gu/facia/api/FAPI.scala

```diff
@@ -87,6 +87,19 @@ object FAPI {
     for(snapContent <- ContentApi.latestContentFromLatestSnaps(capiClient, allSnaps, adjustItemQuery))
       yield snapContent}
 
+  def getTreatsForCollection(collection: Collection, adjustSearchQuery: AdjustSearchQuery = identity, adjustItemQuery: AdjustItemQuery = identity)
+                      (implicit capiClient: GuardianContentClient, ec: ExecutionContext) = {
+    val treatsRequest = Collection.treatsRequestFor(collection)
+    ContentApi.buildHydrateQueries(capiClient, treatsRequest.treatIds, adjustSearchQuery) match {
+      case Success(hydrateQueries) =>
```

Comment:
It seems like there's a bit more logic here than is ideal? The goal is that the FAPI class contains methods that do nothing but trivially thread together Responses in a for-comp. Here we've got nested logic, matching and explicit error handling. This is connected to the fact that the `treatsRequestFor` method is a bit odd as it stands. I think this collectively this is all just 'code smell' that stems from this function trying to do too much stuff.